Userland/Libraries/LibJS/Runtime/Map.h

```diff
@@ -22,13 +24,89 @@ class Map : public Object {
     explicit Map(Object& prototype);
     virtual ~Map() override;
 
-    OrderedHashMap<Value, Value, ValueTraits> const& entries() const { return m_entries; };
-    OrderedHashMap<Value, Value, ValueTraits>& entries() { return m_entries; };
+    void map_clear();
+    bool map_remove(Value const&);
+    Optional<Value> map_get(Value const&) const;
+    bool map_has(Value const&) const;
+    void map_set(Value const&, Value);
+    size_t map_size() const;
```

Comment:
is there any specific reason for prefixing everything with `map_`? we don't usually do that afaik.

Reply:
(and doing `map->map_foo()` everywhere seems weird IMO)

Reply:
Yes, Map is an Object, which has `set`/`get`/etc.

Reply:
Not sure I understand what you're trying to say, I understand it has these methods, I'm asking why they are prefixed with `map_`

Reply:
Because a normal `set()` would shadow the virtual `set()` function?